pkgs/misc/screensavers/pipes-rs/default.nix

```diff
@@ -0,0 +1,26 @@
+{ rustPlatform, fetchFromGitHub, lib }:
+
+rustPlatform.buildRustPackage rec {
+  pname = "pipes-rs";
+  version = "1.4.4";
+
+  src = fetchFromGitHub {
+    owner = "lhvy";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "drqoKkju1EkcWGNnliEah37wVhtU2ddJSOZ5MnCNbuo=";
+  };
+
+  cargoLock = { lockFile = ./Cargo.lock; };
+
+  postPatch = ''
+    cp ${./Cargo.lock} Cargo.lock
+  '';
+
+  meta = with lib; {
+    description = "An over-engineered rewrite of pipes.sh in Rust";
+    homepage = "https://github.com/lhvy/pipes-rs";
+    license = licenses.apsl20;
```

Comment:
`apsl20` is _Apple Public Source License 2.0_, but there are _Apache_ and _MIT_ license files in this project and also in `crates/pipes-rs/Cargo.toml`.
These can be looked up in `lib/licenses.nix`.